rasa/shared/core/flows/flows_yaml_schema.json

```diff
@@ -0,0 +1,293 @@
+          },
+          {
+            "required": [
+              "intent"
+            ],
+            "additionalProperties": false,
+            "properties": {
+              "id": {
+                "type": "string"
+              },
+              "intent": {
+                "type": "string"
+              },
+              "next": {
+                "$ref": "#$defs/next"
+              }
+            }
+          },
+          {
+            "required": [
+              "entry_prompt"
+            ],
+            "additionalProperties": false,
+            "properties": {
+              "id": {
+                "type": "string"
+              },
+              "entry_prompt": {
+                "type": "string"
+              }
+            }
+          }
+        ]
+      }
+    },
+    "flow": {
+      "required": [
+        "steps"
+      ],
+      "type": "object",
+      "additionalProperties": false,
+      "properties": {
+        "description": {
+          "type": "string"
+        },
+        "if": {
+          "type": "string"
+        },
+        "name": {
+          "type": "string"
+        },
+        "nlu_trigger": {
+          "type": "array",
+          "items": {
+            "required": [
+              "intent"
+            ],
+            "type": "object",
+            "properties": {
+              "intent": {
+                "type": "object",
+                "properties": {
+                  "confidence_threshold": {
+                    "type": "number"
+                  },
+                  "name": {
+                    "type": "string"
+                  }
+                },
+                "required": [
+                  "name"
+                ]
+              }
+            }
+          }
+        },
+        "steps": {
+          "$ref": "#$defs/steps"
+        }
+      }
+    },
+    "next": {
+      "anyOf": [
+        {
+          "type": "array",
+          "minContains": 1,
+          "items": {
+            "type": "object",
+            "oneOf": [
+              {
+                "required": [
+                  "if",
+                  "then"
+                ]
+              },
+              {
+                "required": [
+                  "else"
+                ]
+              }
+            ],
+            "properties": {
+              "else": {
+                "oneOf": [
+                  {
+                    "type": "string"
+                  },
+                  {
+                    "$ref": "#$defs/steps"
+                  }
+                ]
+              },
+              "if": {
+                "type": "string"
+              },
+              "then": {
+                "oneOf": [
+                  {
+                    "$ref": "#$defs/steps"
+                  },
+                  {
+                    "type": "string"
+                  }
+                ]
+              }
+            }
+          }
+        },
+        {
+          "type": "string"
```

Comment:
@twerkmeister on the demo llm bot while running e2e tests specifically on [this flow](https://github.com/RasaHQ/financial-demo-flows-llms/blob/main/data/flows/replace_card.yml#L11) 
I get a following schema error:
`On instance['flows']['replace_card']['steps'][0]: `
`('then', [ordereddict([('next', None), ('id', '1_link_replace_eligible_card'), ('link', 'replace_eligible_card')])])])`

is the link step required to have a next and why is it `None` in this case?

Reply:
Actually it's None because `link` steps are disallowed from having `next` steps. So that's something that happens when Flows are instantiated from yaml.

Reply:
That makes sense thanks for the explanation 🙏  made the schema reflect that next can only be `None`.